Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -179,6 +200,12 @@ private bool IsField(FieldInfo field)
                 return true;
             }
 
+            if (!this.options.IgnoreReadOnlyProperties &&
```

Comment:
How is this related to this PR? Seems like maybe this needs to be rebased?

Reply:
This logic is for detecting backing fields for readonly properties since we can't deserialize read-only properties the normal way. 
It's still related to this PR because this PR introduced the `if (IsReadOnlyProperty(property))` check in `Deserialize()`